pkg/function-proxy/proxy.go

```diff
@@ -80,10 +83,35 @@ func startNativeDaemon() {
 	}
 }
 
+func gracefulShutdown(server *http.Server) {
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+	<-stop
+	timeout := 10 * time.Second
```

Comment:
it may be useful to configure this timeout, maybe we should add a env var to be able to change the default value?

Reply:
I've added an env var SHUTDOWN_TIMEOUT to control this.